dnfdaemon-server/services/goal/goal.cpp

```diff
@@ -76,6 +73,31 @@ sdbus::MethodReply Goal::resolve(sdbus::MethodCall & call) {
 
     auto reply = call.createReply();
     reply << result;
+
+    std::vector<dnfdaemon::GoalResolveLogItem> vector_goal_resolve_log_item;
+
+    for (const auto & log : goal.get_resolve_log()) {
+        // TODO(nsella) better use of KeyValueMap with GoalJobSettings
+        dnfdaemon::KeyValueMap goal_job_settings;
+        goal_job_settings.emplace(std::make_pair("to_repo_ids", std::get<2>(log).to_repo_ids));
+
+        vector_goal_resolve_log_item.push_back(dnfdaemon::GoalResolveLogItem(
+                static_cast<uint32_t>(std::get<0>(log)), // action
+                static_cast<uint32_t>(std::get<1>(log)), // problem
+                goal_job_settings, // goal_job_settings
+                std::get<3>(log),  // report
+                std::vector<std::string>{std::get<4>(log).begin(), std::get<4>(log).end()} // vector_report
+                ));
+
+    }
+    dnfdaemon::DbusGoalResolveResults dbus_goal_resolve_results(
+            static_cast<uint32_t>(transaction.get_problems()), // transaction.get_problems()
+            goal.get_formated_all_problems(), // get_formatted_all_problems
```

Comment:
Please check this PR - https://github.com/rpm-software-management/libdnf/pull/1321/files (it already has been merged). The `get_formated_all_problems()` method is now renamed and moved to `Transaction` class.